.github/workflows/validate-models.yml

```diff
@@ -0,0 +1,147 @@
+
+          echo "BOSL2 installed to ./bin/openscad/libraries/BOSL2"
+
+      - name: Find and validate models
+        run: |
+          set -e
+
+          # Convention-based: only test files in specific subdirectories
+          MODEL_DIRS="wallmount flexmount gridfinity rackmount_ears"
+          FAILED_MODELS=()
+          VALIDATED_COUNT=0
+
+          # Use the installed OpenSCAD from bin/openscad
+          OPENSCAD_EXE="./bin/openscad/openscad"
+          LIBRARIES_DIR="./bin/openscad/libraries"
+
+          echo "========================================="
+          echo "Finding OpenSCAD models to validate..."
+          echo "========================================="
+          echo "Using OpenSCAD: ${OPENSCAD_EXE}"
+          echo "Libraries path: ${LIBRARIES_DIR}"
+          echo ""
+
+          for dir in $MODEL_DIRS; do
+            model_path="models/${dir}"
+            if [ -d "${model_path}" ]; then
+              echo "Searching in ${model_path}/"
+              for model in "${model_path}"/*.scad; do
+                if [ -f "${model}" ]; then
+                  echo "  Found: ${model}"
+
+                  # Attempt to render to STL
+                  output_file="/tmp/$(basename ${model} .scad).stl"
+                  echo "  Rendering to ${output_file}..."
+
+                  if OPENSCADPATH="${LIBRARIES_DIR}" \
+                     "${OPENSCAD_EXE}" -o "${output_file}" \
+                     "${model}" --export-format=binstl 2>&1 \
+                     | tee "/tmp/$(basename ${model}).log"; then
+                    # Verify output file exists and is not empty
+                    if [ -f "${output_file}" ] && [ -s "${output_file}" ]
+                    then
+                      file_size=$(stat -c%s "${output_file}" 2>/dev/null \
+                        || stat -f%z "${output_file}" 2>/dev/null)
+                      echo "  ✓ SUCCESS: ${model} (${file_size} bytes)"
+                      VALIDATED_COUNT=$((VALIDATED_COUNT + 1))
+                    else
+                      echo "  ✗ FAILED: ${model} - Output file is empty"
+                      FAILED_MODELS+=("${model}")
+                    fi
+                  else
+                    echo "  ✗ FAILED: ${model} - OpenSCAD rendering error"
+                    FAILED_MODELS+=("${model}")
+                  fi
+
+                  echo ""
+                fi
+              done
+            fi
+          done
+
+          echo "========================================="
+          echo "Validation Results"
+          echo "========================================="
+          echo "Total models validated: ${VALIDATED_COUNT}"
+
+          if [ ${#FAILED_MODELS[@]} -gt 0 ]; then
+            echo "Failed models: ${#FAILED_MODELS[@]}"
+            for failed in "${FAILED_MODELS[@]}"; do
+              echo "  - ${failed}"
+            done
+            exit 1
+          else
+            echo "All models validated successfully! ✓"
+          fi
```

Comment:
it's fine to look for testable files but why not rewrite the testing script to accept filenames as well?